python/ray/dashboard/subprocesses/handle.py

```diff
@@ -137,9 +138,11 @@ def wait_for_module_ready(self):
         Wait for the module to be ready. This is called after start_module()
         and can be blocking.
         """
-        if not self.process_ready_event.wait(
-            dashboard_consts.SUBPROCESS_MODULE_WAIT_READY_TIMEOUT
-        ):
+        if self.parent_conn.poll(dashboard_consts.SUBPROCESS_MODULE_WAIT_READY_TIMEOUT):
```

Comment:
what will happen if child crashes immediately after it starts?

Reply:
Will raise `EOFError` when calling `recv`. Handled in https://github.com/ray-project/ray/pull/52392/commits/742097dca186cd34a796d08adaa37f6063dd97ad